modular_skyrat/modules/moretraitoritems/code/syndiemirror.dm

```diff
@@ -0,0 +1,86 @@
+	new /obj/item/clothing/shoes/chameleon(src)
+	new /obj/item/clothing/glasses/chameleon(src)
+	new /obj/item/clothing/head/chameleon(src)
+	new /obj/item/clothing/mask/chameleon(src)
+	new /obj/item/clothing/neck/chameleon(src)
+	new /obj/item/storage/backpack/chameleon(src)
+	new /obj/item/storage/belt/chameleon(src)
+	new /obj/item/radio/headset/chameleon(src)
+	new /obj/item/stamp/chameleon(src)
+	new /obj/item/pda/chameleon(src)
+	new /obj/item/hhmirror/syndie(src)
+	new /obj/item/dyespray(src)
+
+
+/obj/item/hhmirror/syndie/attack(mob/living/attacked_mob, mob/living/user, params)
+	if(!ishuman(attacked_mob))
+		return
+
+	var/mob/living/carbon/human/target_human = attacked_mob
+
+	var/location = user.zone_selected
+	if(!(location in list(BODY_ZONE_PRECISE_MOUTH, BODY_ZONE_HEAD)) && !user.combat_mode)
+		to_chat(user, span_warning("You stop, look down at what you're currently holding and ponder to yourself, \"This is probably to be used on their hair or their facial hair.\""))
+		return
+
+	if(user.zone_selected != BODY_ZONE_HEAD)
+		return ..()
+
+	var/selected_part = tgui_alert(user, "Please select which part of [target_human] you would like to sculpt!", "It's sculpting time!", list("Hair", "Facial Hair", "Cancel"))
+
+	if(!selected_part || selected_part == "Cancel")
+		return
+
+	if(selected_part == "Hair")
+
+		var/hair_id = tgui_input_list(user, "Please select what hairstyle you'd like to sculpt!", "Select masterpiece", GLOB.hairstyles_list)
+		if(!hair_id)
+			return
+
+		if(hair_id == "Bald")
+			to_chat(target_human, span_danger("[user] seems to be cutting all your hair off!"))
+
+		to_chat(user, span_notice("You begin to masterfully sculpt [target_human]'s hair!"))
+
+		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 10)
+
+		if(do_after(user, haircut_duration, target_human))
+			target_human.hairstyle = hair_id
+			target_human.update_hair()
+			user.visible_message(span_notice("[user] successfully cuts [target_human]'s hair!"), span_notice("You successfully cut [target_human]'s hair!"))
+			new /obj/effect/decal/cleanable/hair(get_turf(src))
+	else
+		var/facial_hair_id = tgui_input_list(user, "Please select what facial hairstyle you'd like to sculpt!", "Select masterpiece", GLOB.facial_hairstyles_list)
+		if(!facial_hair_id)
+			return
+
+		if(facial_hair_id == "Shaved")
+			to_chat(target_human, span_danger("[user] seems to be cutting all your facial hair off!"))
+
+		to_chat(user, "You begin to masterfully sculpt [target_human]'s facial hair!")
+
+		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 10)
+
+		if(do_after(user, facial_haircut_duration, target_human))
+			target_human.facial_hairstyle = facial_hair_id
+			target_human.update_hair()
+			user.visible_message(span_notice("[user] successfully cuts [target_human]'s facial hair!"), span_notice("You successfully cut [target_human]'s facial hair!"))
+			new /obj/effect/decal/cleanable/hair(get_turf(src))
```

Comment:
Probably don't want to leave hair behind either, since this is supposed to be for stealth, and it's a mirror, not a pair of scissors.